starknet_py/contract.py

```diff
@@ -590,51 +595,89 @@ def _make_functions(
+                address=implementation, client=proxy_client
+            )
 
-    async def make_contract(self) -> Contract:
-        return await self._make_contract_recursively(step=1, address=self._address)
+        return Contract(
+            address=proxy_contract.address,
+            abi=contract_class.abi or [],
+            client=proxy_client,
+        )
 
-    async def _make_contract_recursively(
-        self, step: int, address: AddressRepresentation
-    ) -> Contract:
-        if address in self._processed_addresses:
-            raise RecursionError("Proxy cycle detected while resolving proxies")
 
-        if step > self._max_steps:
-            raise RecursionError("Max number of steps exceeded while resolving proxies")
+def prepare_proxy_config(
+    proxy_config: Union[bool, Contract.ProxyConfig]
+) -> Contract.ProxyConfig:
+    if isinstance(proxy_config, bool):
+        if not proxy_config:
+            return {}
+        proxy_config = {}
 
-        code = await self._client.get_code(contract_address=parse_address(address))
-        contract = Contract(
-            address=parse_address(address), abi=code.abi, client=self._client
+    if "max_steps" in proxy_config:
+        warnings.warn(
+            "ProxyConfig.max_steps is deprecated. Contract.from_address always makes at most 1 step.",
+            category=DeprecationWarning,
         )
-        self._processed_addresses.add(address)
+    proxy_checks = [OpenZeppelinProxyCheck(), ArgentProxyCheck()]
+    return {"proxy_checks": proxy_checks, **proxy_config}
 
-        is_proxy = False
-        address = 0
-        for proxy_check in self._proxy_checks:
-            if await proxy_check.is_proxy(contract):
-                is_proxy = True
-                address = await proxy_check.implementation_address(contract)
-                break
 
-        if not is_proxy:
-            return contract
+async def get_class_by_address(address: Address, client: Client) -> DeclaredContract:
+    contract_class_hash = 0
+    try:
+        contract_class_hash = await client.get_class_hash_at(contract_address=address)
+        contract_class = await client.get_class_by_hash(class_hash=contract_class_hash)
+    except ClientError as err:
+        if "is not deployed" in err.message:
+            raise ContractNotFoundError(block_hash="latest") from err
+        raise err
 
-        return await self._make_contract_recursively(
-            address=address,
-            step=step + 1,
-        )
+    return contract_class
+
+
+async def get_implementation_from_proxy(
+    proxy_contract: Contract, proxy_config: Contract.ProxyConfig
+) -> int:
+    implementation, proxy_checks = 0, proxy_config.get("proxy_checks", [])
+    for proxy_check in proxy_checks:
+        implementation = await proxy_check.implementation(proxy_contract)
+        if implementation != 0:
+            break
+
+    if implementation == 0:
+        raise ProxyResolutionError(proxy_checks)
```

Comment:
What if someone provides a contract without a proxy and provides config = True? In this case shouldn't we see that there is no proxy and return abi of the original contract?